src/firefly/java/edu/caltech/ipac/firefly/server/ServCommand.java

```diff
@@ -22,49 +22,39 @@ public void processRequest(HttpServletRequest req, HttpServletResponse res, SrvP
         if (doJsonp) res.setContentType("application/javascript");
 
         String jsonData;
+        JSONObject json=new JSONObject();
+
         try {
             String result = doCommand(new SrvParam(sp.getParamMap()));
 
             if (getCanCreateJson()) {
                 jsonData = result;
             } else {
-                jsonData = "[{" +
-                        "\"success\" :  \"" + true + "\"," +
-                        "\"data\" :  \"" + StringUtils.escapeQuotes(result) + "\"" +
-                        "}]";
-            }
-
-            StringBuilder sb= new StringBuilder(jsonData.length()+10);
-            for(char c : jsonData.toCharArray()) {
-                if (!Character.isISOControl(c) && c>31 && c<127 ) {
-                    sb.append(c);
-                }
+                json.put("success","true" );
+                json.put("data", result );
+                jsonData = "["+json.toJSONString()+"]";
             }
 
-            jsonData= sb.toString();
-
-//            res.setContentType("text/plain");
         } catch (Exception e) {
             e.printStackTrace();
+            json.put("success", "false");
+            json.put("error", e.getMessage());
             StringBuilder sb = new StringBuilder(500);
-            sb.append("[{");
-            sb.append("\"success\" :  \"").append(false).append("\",");
-
-            // need to escape double quotes - otherwise JSON will be messed
-            sb.append("\"error\" : \"").append(e.getMessage().replace("\"", "\\\"")).append("\"");
-
             int cnt = 1;
+            String causeMsg= new String();
             for (Throwable t = e.getCause(); (t != null); t = t.getCause()) {
-                sb.append(", ").append("\"cause").append(cnt).append("\" :  \"").append(t.toString().replace("\"", "\\\"")).append("\"");
+                causeMsg.concat((new Integer(cnt)).toString());
+                causeMsg.concat(":"+t.toString());
```

Comment:
I don't use concat, but I am pretty sure it returns a new string and not modify in-place.  So, causeMsg will be an empty string coming out of this loop.  Also, the logic is not the same.  The modified code does not match the original logic.

Reply:
Yea this should be fixed